### DISCUSSION TITLE
Font Awesome Pro

### DISCUSSION THREAD

**Discussion by ddlidded** [requarks/wiki#2835] [General]
Does anyone know of a way to add the pro icon kit into wiki js? There is a much larger variety of options that are apart of the pro package that I already pay for and would like to use them here as well. Let me know if any of you have experience with this. :)

**Comment by mlwilkerson** [comment]
I'm a developer at Font Awesome who works on infrastructure, like our GraphQL API, to help enable integrations with Font Awesome Pro. I see in [the screen cast here](https://blog.js.wiki/news/2021/wiki-js-3-icon-picker) that there are some Font Awesome Pro styles showing up in this preview of the icon picker coming in 3.0 (the Light and Duotone styles, for example are only available in FA Pro).

I've looked around in this repo for where there might be a branch that includes those newer features, but haven't yet found it. I could offer (at least) feedback, and maybe even some PRs, to help with Font Awesome Pro integration.

I've done a fair bit of related work on our [fa-icon-chooser web component](https://github.com/fortAwesome/fa-icon-chooser) (which _could_ be snapped in here as a bit of pre-made UI), and I've demonstrated an integration pattern in our [official WordPress plugin](https://github.com/FortAwesome/wordpress-fontawesome).

Those integrations enable using the latest-available releases, Pro Kits, and even custom icon uploads in kits.

@NGPixel if you could point me to the code for the 3.x branch, I could take a look and offer suggestions and maybe PRs.

**Reply by NGPixel** [reply]
The 3.x project is available at https://github.com/requarks/wiki-vnext
However, it's not very friendly yet for other devs to look into.

Your component is very interesting and I would definitely like to include it! I see there's a web component and react version. This project is however using Vue 3. Can the web component version be integrated easily?

I don't have a Font Awesome Pro license, can the component still be used without one? or would it be possible to get one for testing? I haven't looked too much into how Font Awesome Pro handle the licensing but I would assume the user would insert a key of some sort into the Wiki.js admin?

**Reply by mlwilkerson** [reply]
Hi @NGPixel,

The `fa-icon-chooser` web component is implemented using [StencilJS](https://stenciljs.com/), which has the ability to build various Framework Integrations, including [Vue](https://stenciljs.com/docs/vue). We also use Vue on `fontawesome.com`, so it seems inevitable that we'd add that Vue integration eventually anyway. It's just a matter of configuring the build in that `fa-icon-chooser` repo to output the Vue component, alongside the React one, and then work out the deployment workflow to publish the NPM.

As for Pro licensing and distribution, yes, the user would add an API Token, which would authorize Wiki.js to issue queries to our GraphQL API on the user's behalf, to retrieve Pro assets, kits, and even their custom icon uploads. As an example, here's some user documentation for how to go about setting up our WordPress plugin with an API Token to use Pro Kits. The flow would presumably be similar in Wiki.js. I can provide more detailed guidance on getting that going if you take that route.

As for getting a license for testing, could you contact me separately by emailing hello at fontawesome dot com? Ask for Mike 😄

**Comment by mrkhachaturov** [comment]
![CleanShot 2025-06-06 at 06 22 25@2x](https://github.com/user-attachments/assets/6c2959f7-2e14-434e-ac3d-b361a1c9c117)
![CleanShot 2025-06-06 at 06 23 46@2x](https://github.com/user-attachments/assets/c3434f26-57d1-4776-a778-09436f1c6687)
